### ISSUE TITLE
Provide a way to use the requirements merging for requirements that don't come from files.

### ISSUE THREAD

**Issue by tomprince** [peterdemin/pip-compile-multi#283]
I have a project that is using [flit](https://flit.readthedocs.io/en/latest/index.html) and [PEP 621](https://flit.readthedocs.io/en/latest/pyproject_toml.html#new-style-metadata) configuration for declaring dependencies. I'd like to be able to use pip-compile-multi to generate minimal compatible requirements files for it.

Currently, I have a script
```shell
pip-compile -q --allow-unsafe pyproject.toml -o requirements/base.txt
pip-compile -q --allow-unsafe pyproject.toml --extra test -o requirements/test.txt
pip-compile -q --allow-unsafe pyproject.toml --extra dev --extra test -o requirements/dev.txt
```
for pinning the requirements, but I'd prefer to use pip-compile-multi in some way, so:
- the requirements files don't have duplication
- there are no conflicts between the generated requirements.

**Comment by peterdemin** [comment]
I'm not using `pyproject.toml` in any of my projects, but I see the value in the standard. Having `pip-compile-multi` parse input requirements from `pyproject.toml` makes sense to me. I can see it as an extra option `--pyproject pyproject.toml`. Would you be interested in exploring this feature?

**Comment by cbenz** [comment]
Yes, absolutely. I don't use flit but followed the [official packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/) which promotes the usage of `pyproject.toml` and declaring dependencies inside it.

I would even suggest that pip-compile-multi read it by default, and then tries to find `requirements/base.in` as a fallback. What do you think about this approach?

Cf https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#dependencies-optional-dependencies

**Comment by lord-haffi** [comment]
I'd be interested in this feature too, thumbs up from me 👍
We are using Hatch, in case you need to know.

**Comment by peterdemin** [comment]
I'm all for this, but don't have time to implement it. I can review/merge a PR and make a release, though.

**Comment by lord-haffi** [comment]
I'm willing to look into this but can't promise to have enough time for this either. So no stress. Btw, great project 😊 Makes it a lot easier to manage multiple environments 👍
